From: Andi Kleen <ak@linux.intel.com>
Date: Mon, 22 Nov 2010 14:05:09 +0000 (15:05 +0100)
Subject: x86: mce: don't print "human readable" message for corrected errors

The "human readable" message is not useful for corrected
errors. There's no reason to run mcelog --ascii and the
error is logged anyways. And it's pr_emerg, which is not
needed for a corrected error.

Only print it during panic.

diff -ur linux-kernel-2.6.38.3/arch/x86/kernel/cpu/mcheck/mce.c linux-kernel-2.6.38.3.new//arch/x86/kernel/cpu/mcheck/mce.c
--- linux-kernel-2.6.38.3/arch/x86/kernel/cpu/mcheck/mce.c	2011-04-15 04:03:56.000000000 +0800
+++ linux-kernel-2.6.38.3.new//arch/x86/kernel/cpu/mcheck/mce.c	2011-07-18 09:58:05.294961747 +0800
@@ -104,11 +104,15 @@
 ATOMIC_NOTIFIER_HEAD(x86_mce_decoder_chain);
 EXPORT_SYMBOL_GPL(x86_mce_decoder_chain);
 
+#define CORRECTED 1
+
 static int default_decode_mce(struct notifier_block *nb, unsigned long val,
 			       void *data)
 {
-	pr_emerg(HW_ERR "No human readable MCE decoding support on this CPU type.\n");
-	pr_emerg(HW_ERR "Run the message through 'mcelog --ascii' to decode.\n");
+	if (val != CORRECTED) { 
+	               pr_emerg(HW_ERR "No human readable MCE decoding support on this CPU type.\n");
+	               pr_emerg(HW_ERR "Run the message through 'mcelog --ascii' to decode.\n");
+	}
 
 	return NOTIFY_STOP;
 }
@@ -588,7 +592,8 @@
 		 */
 		if (!(flags & MCP_DONTLOG) && !mce_dont_log_ce) {
 			mce_log(&m);
-			atomic_notifier_call_chain(&x86_mce_decoder_chain, 0, &m);
+			atomic_notifier_call_chain(&x86_mce_decoder_chain, 
+			            CORRECTED, &m);
 			add_taint(TAINT_MACHINE_CHECK);
 		}
 
